src/Functions/FunctionsLogical.cpp

```diff
@@ -507,10 +508,86 @@ DataTypePtr FunctionAnyArityLogical<Impl, Name>::getReturnTypeImpl(const DataTyp
             : result_type;
 }
 
+template <typename Name>
+static void applyTernaryLogic(const IColumn::Filter & mask, IColumn::Filter & null_bytemap)
+{
+    for (size_t i = 0; i != mask.size(); ++i)
+    {
+        if (null_bytemap[i] && ((Name::name == NameAnd::name && !mask[i]) || (Name::name == NameOr::name && mask[i])))
```

Comment:
Don't we have too many string comparisons here?
Maybe, compare once and write 3 different loops?

Reply:
Make different loops will be better, thanks, will change it